test/e2e/storage/csi_objects.go

```diff
@@ -413,3 +465,61 @@ func deployGCEPDCSIDriver(
 	framework.ExpectNoError(err, "Failed to create DaemonSet: %v", nodeds.Name)
 
 }
+
+func shredFile(filePath string) {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		framework.Logf("File %v was not found, skipping shredding", filePath)
+		return
+	}
+	framework.Logf("Shredding file %v", filePath)
+	_, _, err := framework.RunCmd("shred", "--remove", filePath)
+	if err != nil {
+		framework.Logf("Failed to shred file %v: %v", filePath, err)
+	}
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		framework.Logf("File %v successfully shredded", filePath)
+		return
+	}
+	// Shred failed Try to remove the file for good meausure
+	err = os.Remove(filePath)
+	framework.ExpectNoError(err, "Failed to remove service account file %s", filePath)
+
+}
+
+// createGCESecrets downloads the GCP IAM Key for the default compute service account
+// and puts it in a secret for the GCE PD CSI Driver to consume
+func createGCESecrets(client clientset.Interface, config framework.VolumeTestConfig) {
+	saEnv := "E2E_GOOGLE_APPLICATION_CREDENTIALS"
+	saFile := "/tmp/cloud-sa.json"
+
+	premadeSAFile, ok := os.LookupEnv(saEnv)
+	if !ok {
+		framework.Logf("Could not find env var %v, please either create cloud-sa"+
+			" secret manually or rerun test after setting %v to the filepath of"+
+			" the GCP Service Account to give to the GCE Persistent Disk CSI Driver", saEnv, saEnv)
+		return
+	}
+
+	framework.Logf("Found CI service account key at %v", premadeSAFile)
+	// Need to copy it saFile
+	stdout, stderr, err := framework.RunCmd("cp", premadeSAFile, saFile)
```

Comment:
do we need to copy it? are we modifying it?

Reply:
no, but it needs to be called `cloud-sa.json` and this is the cleanest way of doing that. Another method could be to change the testing manifest to use whatever the file in `E2E_GOOGLE_APPLICATION_CREDENTIALS` but thats arguably more hacky as it requires digging into the manifest, looping through its containers, looping through the right containers env vars and changing the value of the `GOOGLE_APPLICATION_CREDENTIALS` variable.